src/rate_limiter/src/lib.rs

```diff
@@ -347,33 +358,64 @@ impl RateLimiter {
         })
     }
 
+    // Arm the timer of the rate limiter with the provided `TimerState`.
+    fn activate_timer(&mut self, timer_state: TimerState) {
+        // Register the timer; don't care about its previous state
+        self.timer_fd.set_state(timer_state, SetTimeFlags::Default);
+        self.timer_active = true;
+    }
+
     /// Attempts to consume tokens and returns whether that is possible.
     ///
     /// If rate limiting is disabled on provided `token_type`, this function will always succeed.
     pub fn consume(&mut self, tokens: u64, token_type: TokenType) -> bool {
+        // If the timer is active, we can't consume tokens from any bucket and the function fails.
+        if self.timer_active {
+            return false;
+        }
+
         // Identify the required token bucket.
         let token_bucket = match token_type {
             TokenType::Bytes => self.bandwidth.as_mut(),
             TokenType::Ops => self.ops.as_mut(),
         };
         // Try to consume from the token bucket.
-        let success = match token_bucket {
-            Some(bucket) => bucket.reduce(tokens),
+        match token_bucket {
```

Comment:
`if let` would make it more readable. Nested matches usually are hard to read.

Reply:
Changed it to an if-let statement.